sdk/storage/AzureStorageBlob/Source/DataStructures/URLSessionTransferManager.swift

```diff
@@ -0,0 +1,389 @@
+            }
+        }
+
+        // Remove the blob operation from the transfers list
+        if let index = transfers.firstIndex(where: { $0 === transfer }) {
+            transfers.remove(at: index)
+        }
+
+        // remove the object from CoreData which should cascade and delete any outstanding block transfers
+        if let context = persistentContainer?.viewContext {
+            context.delete(transfer)
+        }
+    }
+
+    // MARK: Pause Operations
+
+    public func pause(transfer: Transfer) {
+        let allowed: [TransferState] = [.pending, .inProgress]
+        guard allowed.contains(transfer.state) else { return }
+
+        switch transfer {
+        case let transfer as BlockTransfer:
+            pause(transfer: transfer)
+        case let transfer as BlobTransfer:
+            pause(transfer: transfer)
+        default:
+            fatalError("Unrecognized transfer type: \(transfer.self)")
+        }
+        transfer.state = .paused
+        operation(transfer.operation, didChangeState: transfer.state)
+    }
+
+    internal func pause(transfer: BlockTransfer) {
+        assert(transfer.operation != nil, "Transfer operation unexpectedly nil.")
+        if let operation = transfer.operation {
+            operationQueue.pause(operation)
+        }
+    }
+
+    internal func pause(transfer: BlobTransfer) {
+        // Cancel the operation and any associated block operations
+        if let operation = transfer.operation {
+            operationQueue.remove(operation)
+            // transfer.state = .paused
+            for block in transfer.transfers {
+                block.state = .paused
+                if let blockOp = block.operation {
+                    operationQueue.remove(blockOp)
+                }
+            }
+        }
+    }
+
+    // MARK: Resume Operations
+
+    public func resume(transfer: Transfer) {
+        guard reachability?.isReachable ?? false else { return }
+        let allowed: [TransferState] = [.paused]
+        guard allowed.contains(transfer.state) else { return }
+        transfer.state = .pending
+
+        switch transfer {
+        case let transfer as BlockTransfer:
+            operationQueue.add(BlockOperation(withTransfer: transfer))
+        case let transfer as BlobTransfer:
+            for blockTransfer in transfer.transfers {
+                if allowed.contains(blockTransfer.state) {
+                    blockTransfer.state = .pending
+                }
+            }
+            queueOperationsFor(blobTransfer: transfer)
+        default:
+            fatalError("Unrecognized transfer type: \(transfer.self)")
+        }
+        operation(transfer.operation, didChangeState: transfer.state)
+    }
+
+    // MARK: Core Data Operations
+
+    public func loadContext() {
+        // Hydrate operationQueue from CoreData
+        guard let context = persistentContainer?.viewContext else {
+            fatalError("Unable to load persistent container.")
```

Comment:
When would this condition fail? And is there something better to do than a fatalError?

Reply:
This would fail if the persistent container were nil, because the bundle identifier or something else was wrong. Arguably it could proceed as if there was no saved CoreData